demo.cc

```diff
@@ -276,11 +313,13 @@ class Cruncher final {
     inline void operator()(schema::UserRecord& u) {
       std::cerr << '@' << demo_id_ << " +U: " << u.uid << '\n';
       box_.users.push_back(u.uid);
+      time_window_tracker_.Add(static_cast<double>(u.ms));
```

Comment:
`ExtractTimestamp()` (or something of similar name) is for Sherlock to use the stream. Used for further joins of multiple streams. The default implementation indeed accesses the `.ms` field, but it doesn't have to.